### ISSUE TITLE
AttachTo doesn't find elements in ember-wormhole

### ISSUE THREAD

**Issue by BitBrit** [shipshapecode/ember-shepherd#343]
Hi, 

I am trying to create a tour, some of which goes through an ember-bootstrap modal. Ember-bootstrap modals use ember-wormhole. 

I was hoping that I would be able to use the attachTo property and it would find elements and place the tooltip accordingly but unfortunately I get the console error stating that the element in the modal can't be found so it just places the tool tip in the center of the viewport. 

Is there any work around/technique that I can use to make this work or is this just completely not supported?

Thanks in advance.

**Comment by rwwagner90** [comment]
Hi @BitBrit it should support targeting any elements in the DOM. Do you have a small reproduction we could play with?

**Comment by BitBrit** [comment]
I don't yet, but let me see what I can put together. Actually, knowing that it _should_ be able to target these elements helps.

**Comment by rwwagner90** [comment]
@BitBrit I do not know what wormhole does, but any elements under `body` should work.

**Comment by BitBrit** [comment]
Ok, then perhaps it's a timing thing. It could be looking for the element in the modal before the modal has finished rendering.

**Comment by BitBrit** [comment]
Hi @rwwagner90 Thank you for your prompt responses. My suggestion above was correct. I had to find a way to advance the tour only when the modal had completed it's rendering.  This means "advanceOn" wasn't usable without creating some other special event for it to listen to.

**Comment by rwwagner90** [comment]
@BitBrit glad you figured it out! I typically suggest not to use `advanceOn` and to manually call `tour.next`, so you can fully control everything.

**Comment by BitBrit** [comment]
That sounds good but the reason I used `advanceOn` was because I didn't want the tour logic/knowledge to bleed throughout the app. But yes, it seems to have its short-comings.

**Comment by rwwagner90** [comment]
@BitBrit Sure, that makes sense. I'm open to suggestions to making `advanceOn` better, but I think making a custom event that is sure the element is visible might be the best option. We could also maybe have Shepherd continue to try to find the element to attach to even like 500ms or something.

**Comment by BitBrit** [comment]
@rwwagner90 that would be really good.

We are also using ember-bootstrap with a dropdown-menu element. Our tour spans this dropdown menu and so we have to wait for it to exist. ember-bootstrap usually has a way to pass an `onShown()` action handler. However, for dropdown menus, it does not.  This meant that we had to develop our own way of trying to detect when the menu has finished opening and is in the DOM. 

We thought about adding some sort of timeout after which we would know for certain the menu had finished transitioning to being open but this felt a little "hacky". So, rightly or wrongly, we decided to use this: 

https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver

So that it doesn't become a performance problem we scoped the conditions as to when this would fire as tightly as possible.  Your suggestion would mean that we wouldn't have to implement things like this to handle transitions.
